### DISCUSSION TITLE
Getting excpetion in Azure redis.clients.jedis.exceptions.JedisConnectionException: java.net.SocketException: Connection or outbound has closed

### DISCUSSION THREAD

**Discussion by G-ruchika** [xetorthio/jedis#3316] [General]
I am using Jedis client 4.1.0 in my spring boot application. I have deployed my app to azure and it's working fine for some time and then after 1 day of deploy getting the exception.  **`redis.clients.jedis.exceptions.JedisConnectionException: java.net.SocketException: Connection or outbound has closed`**

Defined the Jedis connection details in application yml  and creating the Jedis client like this .

new Jedis(hostName, port, DefaultJedisClientConfig.builder()
                .password(password)
                .ssl(useSsl)
                .build());  

Any suggestion how i can fix this issue?  
Thanks!.

**Comment by sazzad16** [comment]
@G-ruchika First of all, please do not post repeated posts.

It seems Redis server closed the connection. But it should happen only for idle connections.

Please post detailed stacktrace. Also, please confirm whether it is a thrown a exception or just an error log.

**Reply by G-ruchika** [reply]
@sazzad16 Sorry about that. It happen accidentally and could not delete it. I do not have full stack trace as we are not catching the exception to bubble up but it's thrown exception redis.clients.jedis.exceptions.JedisConnectionException: java.net.SocketException: Connection or outbound has closed.

**Comment by sazzad16** [comment]
As you are using SSL, you may try newly generated certs. You can also check the SSLParameters.

**Reply by G-ruchika** [reply]
Yes. i am sending SSL true but i am not using any certs. i am using connection string and password.

**Reply by sazzad16** [reply]
You should/must use certs.
